Mathlib/Tactic/DeriveTraversable.lean

```diff
@@ -0,0 +1,459 @@
+/-
+Copyright (c) 2018 Simon Hudon. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Simon Hudon
+
+! This file was ported from Lean 3 source module control.traversable.derive
+! leanprover-community/mathlib commit 70fd9563a21e7b963887c9360bd29b2393e6225a
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.Tactic.Basic
+import Mathlib.Control.Traversable.Lemmas
+
+/-!
+## Automation to construct `Traversable` instances
+-/
+
+namespace Mathlib.Deriving.Traversable
+
+open Lean Meta Elab Term Command Tactic Match List Monad Functor
+
+/-- similar to `nestedTraverse` but for `Functor` -/
+partial def nestedMap (f v t : Expr) : TermElabM Expr := do
+  let t ← instantiateMVars t
+  if ← withNewMCtxDepth <| isDefEq t v then
+    return f
+  else if !v.occurs t.appFn! then
+    let cl ← mkAppM ``Functor #[t.appFn!]
+    let inst ← synthInstance cl
+    let f' ← nestedMap f v t.appArg!
+    mkAppOptM ``Functor.map #[t.appFn!, inst, none, none, f']
+  else throwError "type {t} is not a functor with respect to variable {v}"
+
+/-- similar to `traverseField` but for `Functor` -/
+def mapField (n : Name) (cl f α β e : Expr) : TermElabM Expr := do
+  let t ← whnf (← inferType e)
+  if t.getAppFn.constName = some n then
+    throwError "recursive types not supported"
+  else if α.eqv e then
+    return β
+  else if α.occurs t then
+    let f' ← nestedMap f α t
+    return f'.app e
+  else if ←
+      (match t with
+        | .app t' _ => withNewMCtxDepth <| isDefEq t' cl
+        | _ => return false) then
+    mkAppM ``Comp.mk #[e]
+  else
+    return e
+
+def getAuxDefOfDeclName : TermElabM FVarId := do
```

Comment:
Unfortunately the docBlame linter was temporarily disabled during the port. Nevertheless, could you make sure that #lint is happy?